primitives/src/merkle_tree/internal.rs

```diff
@@ -226,6 +231,90 @@ where
     }
 }
 
+#[allow(clippy::type_complexity)]
+pub(crate) fn build_light_weight_tree_internal<E, H, I, Arity, T>(
+    height: usize,
+    elems: impl IntoIterator<Item = impl Borrow<E>>,
+) -> Result<(Box<MerkleNode<E, I, T>>, u64), PrimitivesError>
+where
+    E: Element,
+    H: DigestAlgorithm<E, I, T>,
+    I: Index + From<u64>,
+    Arity: Unsigned,
+    T: NodeValue,
```

Comment:
What's the difference from `NodeValue` and `Element`? Is that `Element` has a uid?

Reply:
`NodeValue` is the internal Merkle node value. `Element` is the type of the stored element.